openedx/core/djangoapps/user_api/preferences/tests/test_views.py

```diff
@@ -0,0 +1,484 @@
+        self.send_patch(self.client, {
+            "dict_pref": {"int_key": 10},
+            "string_pref": "value",
+        })
+        response = self.send_get(self.client)
+        self.assertEqual({u"dict_pref": u"{u'int_key': 10}", u"string_pref": u"value"}, response.data)
+
+    @ddt.data(
+        ("different_client", "different_user"),
+        ("staff_client", "staff_user"),
+    )
+    @ddt.unpack
+    def test_create_preferences_other_user(self, api_client, user):
+        """
+        Test that a client (logged in) cannot create preferences for another user.
+        """
+        client = self.login_client(api_client, user)
+        self.send_patch(
+            client,
+            {
+                "dict_pref": {"int_key": 10},
+                "string_pref": "value",
+            },
+            expected_status=404,
+        )
+
+    def test_update_preferences(self):
+        """
+        Test that a client (logged in) can update her own preferences information.
+        """
+        # Create some test preferences values.
+        UserPreference.set_preference(self.user, "dict_pref", {"int_key": 10})
+        UserPreference.set_preference(self.user, "string_pref", "value")
+        UserPreference.set_preference(self.user, "extra_pref", "extra_value")
+
+        # Send the patch request
+        self.client.login(username=self.user.username, password=self.test_password)
+        self.send_patch(self.client, {
+            "string_pref": "updated_value",
+            "new_pref": "new_value",
+            "extra_pref": None,
+        })
+
+        # Verify that GET returns the updated preferences
+        response = self.send_get(self.client)
+        expected_preferences = {
+            "dict_pref": "{'int_key': 10}",
+            "string_pref": "updated_value",
+            "new_pref": "new_value",
+        }
+        self.assertEqual(expected_preferences, response.data)
+
+    def test_update_preferences_bad_data(self):
+        """
+        Test that a client (logged in) receives appropriate errors for a bad update.
+        """
+        # Create some test preferences values.
+        UserPreference.set_preference(self.user, "dict_pref", {"int_key": 10})
+        UserPreference.set_preference(self.user, "string_pref", "value")
+        UserPreference.set_preference(self.user, "extra_pref", "extra_value")
+
+        # Send the patch request
+        self.client.login(username=self.user.username, password=self.test_password)
+        response = self.send_patch(
+            self.client,
+            {
+                "string_pref": "updated_value",
+                TOO_LONG_PREFERENCE_KEY: "new_value",
+                "new_pref": "new_value",
+            },
+            expected_status=400
+        )
+        expected_message = u"The user preference has the following errors: " \
+                           "{'key': [u'Ensure this value has at most 255 characters (it has 256).']}"
```

Comment:
Here it is! And it is indeed ugly! But what is interesting is the "key" part. I suppose that adds some value.

So I am OK with you changing how account validation error messages print out. But I still think we want shared code.